### PR TITLE
[ISSUE #5851] Sandbox API debugging, the target API method is not POST, and the response format is JSON. The API returns an exception.

### PR DESCRIPTION
**Pull request by ISKonst**
…get method is not a POST request.

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java [hunk 1]
```diff
@@ -220,15 +221,16 @@ public String request(final String url, final Map<String, ?> form, final Map<Str
      * @param url    url
      * @param json   json
      * @param header header
+     * @param method method
      * @return Response
      * @throws IOException IOException
      */
     public Response requestJson(final String url, final String json,
-        final Map<String, String> header) throws IOException {
+        final Map<String, String> header, HTTPMethod method) throws IOException {
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java:229]: where is call this funcation？

**Comment by ISKonst** [comment]
SKonst
***@***.***



&nbsp;
HttpUtils#requestJson,This method is currently invoked in only one place.
The call stack of this method is as follows and the source code location..
org.apache.shenyu.admin.controller.SandboxController#proxyGateway. Line 52 - 55
&nbsp; &nbsp; org.apache.shenyu.admin.service.impl.SandboxServiceImpl#requestProxyGateway. Line 79 - 110
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;org.apache.shenyu.admin.utils#requestCall. Line 303 - 308
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;org.apache.shenyu.admin.utils#requestJson. Line 226 - 231


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/shenyu"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月20日(星期五) 上午10:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [apache/shenyu] [ISSUE #5851] Sandbox API debugging, the target API method is not POST, and the response format is JSON. The API returns an exception. (PR #5852)





 
@yu199195 commented on this pull request.
 
 
In shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java:
 &gt;       * @return Response       * @throws IOException IOException       */      public Response requestJson(final String url, final String json, -        final Map<String, String&gt; header) throws IOException { +        final Map<String, String&gt; header, HTTPMethod method) throws IOException {  
where is call this funcation？

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java (hunk 1 above)
Comment (ISKonst) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java:229]: [HttpUtils](https://github.com/apache/shenyu/pull/5852#discussion_r1893368809)#requestJson,This method is currently invoked in only one place.
The call stack of this method is as follows and the source code location..
org.apache.shenyu.admin.controller.SandboxController#proxyGateway. Line 52 - 55
    org.apache.shenyu.admin.service.impl.SandboxServiceImpl#requestProxyGateway. Line 79 - 110
        org.apache.shenyu.admin.utils.HttpUtils#requestCall. Line 303 - 308
            org.apache.shenyu.admin.utils.HttpUtils#requestJson. Line 226 - 231

Comment (yu199195) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java:229]: You changed the method signature, don't you need to change it in the call?

Comment (yu199195) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java:229]: Hi，pls check ci，it not passed

Comment (ISKonst) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/utils/HttpUtils.java]: Applied Checkstyle to HttpUtils file

**Review by yu199195 — APPROVED** [review]

**Pull request merged**